test/fixtures/dialects/trino/regexp_replace_with_lambda.sql

```diff
@@ -0,0 +1,8 @@
+-- The variant of REGEXP_REPLACE that accepts a function
+-- lambda expression as an argument can be tricky to parse.
+-- Signature:
+--    regexp_replace(string, pattern, function) → varchar
+-- Reference:
+--     https://trino.io/docs/422/functions/regexp.html
+
+SELECT REGEXP_REPLACE('new york', '(\w)(\w*)', x -> UPPER(x[1]) || LOWER(x[2]));
```

Comment:
nit: please add a newline here

Reply:
I think the newline has now been added, though it's a bit hard to verify in the GitHub UI.